flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/table/IndexGeneratorTest.java

```diff
@@ -101,9 +105,28 @@
                                     LocalDateTime.of(2020, 3, 19, 12, 22, 14, 1000)),
                             (int) LocalDate.of(2020, 3, 19).toEpochDay(),
                             (int) (LocalTime.of(12, 13, 14, 2000).toNanoOfDay() / 1_000_000L),
+                            TimestampData.fromTimestamp(Timestamp.valueOf("2020-03-19 12:22:14")),
                             "test2",
                             false));
 
+    @Test
+    public void testDynamicIndexFromTimestampTz() {
+        IndexGenerator indexGenerator =
+                IndexGeneratorFactory.createIndexGenerator(
+                        "{local_timestamp|yyyy_MM_dd_HH-ss}_index", fieldNames, dataTypes);
+        indexGenerator.open();
+
+        if (ZoneId.systemDefault().equals(ZoneId.of("UTC"))) {
```

Comment:
This if/else doesn't make sense for the test, you need to provide custom timezones for the data.